src/main/java/uk/gov/hmcts/reform/pip/data/management/service/LocationService.java

```diff
@@ -293,6 +279,51 @@ public List<Integer> getAllLocationsWithNamePrefix(String prefix) {
             .toList();
     }
 
+    private void validateLocationName(List<LocationCsv> locationCsvList) {
+        checkLocationNameDuplication(locationCsvList);
+        locationCsvList.forEach(inputLocation -> {
+            try {
+                validationService.containsHtmlTag(inputLocation.getLocationName(),
+                        inputLocation.getWelshLocationName());
+            } catch (ContainsForbiddenValuesException e) {
+                String message = String.format("Failed to upload locations. The location name '%s' or Welsh location "
+                        + "name '%s' contains a forbidden character", inputLocation.getLocationName(),
+                        inputLocation.getWelshLocationName());
+                log.error(writeLog(message));
+                throw new LocationNameValidationException(message);
+            }
+        });
+    }
+
+    private void checkLocationNameDuplication(List<LocationCsv> locationCsvList) {
```

Comment:
I think we need to add some check within the file itself too. E.g. if someone where to add two courts with the same name, within the same file upload, I think based on this it would allow it.

Keeping the constraint validation should stop this, however might be better to do it like the below to get better error messages

Reply:
Added the logic to check location name duplication within the CSV file